x-pack/plugins/reporting/server/deprecations/index.ts

```diff
@@ -19,10 +19,12 @@ export const registerDeprecations = ({
 }) => {
   core.deprecations.registerDeprecations({
     getDeprecations: async (ctx) => {
-      return [
-        ...(await getIlmPolicyDeprecationsInfo(ctx, { reportingCore })),
-        ...(await getReportingRoleDeprecationsInfo(ctx, { reportingCore })),
-      ];
+      return reportingCore.getConfig().statefulSettings.enabled
+        ? [
+            ...(await getIlmPolicyDeprecationsInfo(ctx, { reportingCore })),
+            ...(await getReportingRoleDeprecationsInfo(ctx, { reportingCore })),
+          ]
+        : [];
```

Comment:
Same as above - I think we no longer need this change since the deprecations internally will be looking at the right values in the config